docs/ref/request-response.txt

```diff
@@ -159,14 +162,35 @@ All attributes should be considered read-only, unless stated otherwise.
     converting all characters to uppercase, replacing any hyphens with
     underscores and adding an ``HTTP_`` prefix to the name. So, for example, a
     header called ``X-Bender`` would be mapped to the ``META`` key
-    ``HTTP_X_BENDER``.
+    ``HTTP_X_BENDER``. See also :attr:`~HttpRequest.headers`.
 
     Note that :djadmin:`runserver` strips all headers with underscores in the
     name, so you won't see them in ``META``. This prevents header-spoofing
     based on ambiguity between underscores and dashes both being normalizing to
     underscores in WSGI environment variables. It matches the behavior of
     Web servers like Nginx and Apache 2.4+.
 
+.. _headers:
+.. attribute:: HttpRequest.headers
+
+    A case-insensitive dictionary containing headers passed by the client.
+    Header case is preserved when turning it into a dictionary,
+    or iterating over it (you'll see the original header sent).
+    But when accessing elements with ``headers.get``, ``headers[]`` or
+    checking membership (``in``), case is ignored. Examples::
```

Comment:
Can you explain this? (I'm not sure I quite follow it.) 

* Not matter what the client sends, WSGI gives us headers of the form `HTTP_USER_AGENT` right? 
* These are then transformed in `parse_cgi_header` to hyphened-title-case (is there a name for that?) such as, `User-Agent` (`return cgi_header.replace('_', '-').title()`)
* So in what sense is header case preserved or we "see the original header sent"?

Reply:
You're totally right. I didn't choose the correct words.